storage/pebble/bootstrap_test.go

```diff
@@ -0,0 +1,249 @@
+			require.Equal(t, val, []byte{defaultRegisterValue})
+		}
+
+		require.NoError(t, pb.Close())
+		require.NoError(t, os.RemoveAll(dbDir))
+	})
+}
+
+func TestRegisterBootstrap_IndexCheckpointFile_Empty(t *testing.T) {
+	t.Parallel()
+	log := zerolog.New(io.Discard)
+	rootHeight := uint64(10000)
+	unittest.RunWithTempDir(t, func(dir string) {
+		tries := []*trie.MTrie{trie.NewEmptyMTrie()}
+		fileName := "empty-checkpoint"
+		require.NoErrorf(t, wal.StoreCheckpointV6Concurrently(tries, dir, fileName, log), "fail to store checkpoint")
+		checkpointFile := path.Join(dir, fileName)
+		pb, dbDir := createPebbleForTest(t)
+
+		bootstrap, err := NewRegisterBootstrap(pb, checkpointFile, rootHeight, log)
+		require.NoError(t, err)
+		err = bootstrap.IndexCheckpointFile(context.Background())
+		require.NoError(t, err)
+
+		// create registers instance and check values
+		reg, err := NewRegisters(pb)
+		require.NoError(t, err)
+
+		require.Equal(t, reg.LatestHeight(), rootHeight)
+		require.Equal(t, reg.FirstHeight(), rootHeight)
+
+		require.NoError(t, pb.Close())
+		require.NoError(t, os.RemoveAll(dbDir))
+	})
+}
+
+func TestRegisterBootstrap_IndexCheckpointFile_FormatIssue(t *testing.T) {
+	t.Parallel()
+	pa1 := testutils.PathByUint8(0)
+	pa2 := testutils.PathByUint8(1)
+	rootHeight := uint64(666)
+	pl1 := testutils.LightPayload8('A', 'A')
+	pl2 := testutils.LightPayload('B', 'B')
+	paths := []ledger.Path{pa1, pa2}
+	payloads := []ledger.Payload{*pl1, *pl2}
+	emptyTrie := trie.NewEmptyMTrie()
+	trieWithInvalidEntry, _, err := trie.NewTrieWithUpdatedRegisters(emptyTrie, paths, payloads, true)
+	require.NoError(t, err)
+	log := zerolog.New(io.Discard)
+
+	unittest.RunWithTempDir(t, func(dir string) {
+		fileName := "invalid-checkpoint"
+		require.NoErrorf(t, wal.StoreCheckpointV6Concurrently([]*trie.MTrie{trieWithInvalidEntry}, dir, fileName, log),
+			"fail to store checkpoint")
+		checkpointFile := path.Join(dir, fileName)
+		pb, dbDir := createPebbleForTest(t)
+
+		bootstrap, err := NewRegisterBootstrap(pb, checkpointFile, rootHeight, log)
+		require.NoError(t, err)
+		err = bootstrap.IndexCheckpointFile(context.Background())
+		require.ErrorContains(t, err, "unexpected ledger key format")
+		require.NoError(t, pb.Close())
+		require.NoError(t, os.RemoveAll(dbDir))
+	})
+
+}
+
+func TestRegisterBootstrap_IndexCheckpointFile_CorruptedCheckpointFile(t *testing.T) {
+	t.Parallel()
+	rootHeight := uint64(666)
+	log := zerolog.New(io.Discard)
+	unittest.RunWithTempDir(t, func(dir string) {
+		tries, _ := largeTrieWithValidRegisterIDs(t)
+		checkpointFileName := "large-checkpoint-incomplete"
+		require.NoErrorf(t, wal.StoreCheckpointV6Concurrently(tries, dir, checkpointFileName, log), "fail to store checkpoint")
+		// delete 2nd part of the file (2nd subtrie)
+		fileToDelete := path.Join(dir, fmt.Sprintf("%v.%03d", checkpointFileName, 2))
+		err := os.RemoveAll(fileToDelete)
+		require.NoError(t, err)
+		pb, _ := createPebbleForTest(t)
```

Comment:
don't forget to cleanup the dir (otherwise the collect on people's laptops after running tests)